cmake/Modules/legate_core_options.cmake

```diff
@@ -0,0 +1,125 @@
+#=============================================================================
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+option(BUILD_SHARED_LIBS "Build legate.core shared libraries" ON)
+
+function(set_or_default var_name var_env)
+  list(LENGTH ARGN num_extra_args)
+  if(num_extra_args GREATER 0)
+    list(GET ARGN 0 var_default)
+  endif()
+  if(DEFINED ${var_name})
+    message(VERBOSE "legate.core: ${var_name}=${${var_name}}")
+  elseif(DEFINED ENV{${var_env}})
+    set(${var_name} $ENV{${var_env}} PARENT_SCOPE)
+    message(VERBOSE "legate.core: ${var_name}=$ENV{${var_env}} (from envvar '${var_env}')")
+  elseif(DEFINED var_default)
+    set(${var_name} ${var_default} PARENT_SCOPE)
+    message(VERBOSE "legate.core: ${var_name}=${var_default} (from default value)")
+  else()
+    message(VERBOSE "legate.core: not setting ${var_name}")
+  endif()
+endfunction()
+
+# Initialize these vars from the CLI, then fallback to an envvar or a default value.
+set_or_default(Legion_SPY USE_SPY OFF)
+set_or_default(Legion_USE_LLVM USE_LLVM OFF)
+set_or_default(Legion_USE_CUDA USE_CUDA OFF)
+set_or_default(Legion_USE_HDF5 USE_HDF OFF)
+set_or_default(Legion_USE_GASNet USE_GASNET OFF)
+set_or_default(Legion_USE_OpenMP USE_OPENMP OFF)
+set_or_default(Legion_BOUNDS_CHECKS CHECK_BOUNDS OFF)
+
+option(Legion_SPY "Enable detailed logging for Legion Spy" OFF)
+option(Legion_USE_LLVM "Use LLVM JIT operations" OFF)
+option(Legion_USE_HDF5 "Enable support for HDF5" OFF)
+option(Legion_USE_CUDA "Enable Legion support for the CUDA runtime" OFF)
+option(Legion_USE_GASNet "Enable the distributed GASNet backend" OFF)
+option(Legion_USE_OpenMP "Use OpenMP" OFF)
+option(Legion_USE_Python "Use Python" OFF)
+option(Legion_BOUNDS_CHECKS "Enable bounds checking in Legion accessors" OFF)
+
+if(Legion_USE_GASNet)
+  set_or_default(GASNet_ROOT_DIR GASNET)
+  set_or_default(GASNet_CONDUIT CONDUIT "udp")
+
+  if(Legion_USE_GASNet AND (NOT GASNet_ROOT_DIR))
+    option(Legion_EMBED_GASNet "Embed a custom GASNet build into Legion" ON)
+  endif()
+endif()
+
+set_or_default(Legion_MAX_DIM LEGION_MAX_DIM 4)
```

Comment:
TODO comment since I'm still trying to understand the behavior... If you download legion (rather that pointing to an existing Legion) you get `Legion_MAX_DIM = 3` in `CMakeCache.txt` after running, i.e. the Legion default.  What's so bizarre about this is that I still get `define LEGION_MAX_DIM 4` in `legion_defines.h`.

If run cmake configure again with the pre-existing CMakeCache.txt, the `Legion_MAX_DIM =3` overwrites everything and now you get `define LEGION_MAX_DIM 3` in `legion_defines.h`. 

So basically cmake configure is not "idempotent". If you configure and the configure again without changing anything, you get different header files. Anyhow, I need to look at `rapids_cpm_find` more to understand the behavior here as to why we are not getting `Legion_MAX_DIM=4` in CMakeCache.txt.

Reply:
Fixed in latest PR merge. See https://github.com/trxcllnt/legate.core/pull/2